packages/synapse-react-client/src/utils/context/SynapseContext.tsx

```diff
@@ -50,6 +54,16 @@ export function SynapseContextProvider(props: SynapseContextProviderProps) {
     [providedContext.accessToken],
   )
 
+  const synapseApiClient = useMemo(() => {
+    const configuration = new Configuration({
+      fetchApi: fetchWithExponentialTimeout,
```

Comment:
Super fortunate that our existing abstraction just plugs in here. We pass in our `fetchWithExponentialTimeout` and we maintain the same behavior that we already have with automatic retries and all errors converted to our custom `SynapseClientError` class.